CRISTALINT-13: Offer a way to check if the authentication service is set properly

pjeanjean opened the pull request:
# Jira URL

<!-- Add the link to the corresponding JIRA issue referenced in a commit message. Unless this is a [Misc] commit,
see https://dev.xwiki.org/xwiki/bin/view/Community/DevelopmentPractices#HRule:Don27tcreateunnecessaryissues
-->
https://jira.xwiki.org/browse/CRISTALINT-13

Depends on https://github.com/xwiki-contrib/cristal/pull/1403

# Changes

## Description

<!-- Describe the main changes brought in this PR. -->
Switch the current auth service during REST endpoint initialization.

## Clarifications

<!-- Provide extra hints to make it easier to understand the PR. Those could be:
* Explanation of choices made in this PR
* Anchor towards extra resources needed to understand the context of this PR (e.g., a forum proposal).
* Links to other issues this issue depends on
-->

N/A

# Screenshots & Video

<!-- If this PR introduces any UI change, it's recommended to highlight it with before/after screenshots 
or even a screen recording for complex interactions. 
-->
N/A

# Executed Tests

<!-- Especially important for regression fixes. 
Indicate how changes were tested (e.g., what maven commands were run to validate them).
-->
N/A

# Expected merging strategy

* Prefers squash: Yes <!-- No — Explain why. -->
* Backport on branches:
  * N/A

pjeanjean committed:
CRISTALINT-13: Automatically set authentication service

manuelleduc commented:
I'm not sure if I agree with the proposed solution.
Since Cristal is installed *inside* XWiki. Just using the same cookie should be enough to be considered authenticated.
Imo, we we need to do is:
1. remove the dependency to oidc 
2. provide a dedicated authentication components that
  * check if the users is currently authenticated and if not, provide an authentication form. 
  * add no authentication head since the cookie should be enough

pjeanjean commented:
There are actually 2 issues:
- 1: when accessing Cristal through XWiki, so using [cristal-integration-resource-handler](https://github.com/xwiki-contrib/cristal-integration/tree/main/cristal-integration-resource-handler), there is indeed no need for the OIDC authentication service. Right now, even if it's not enabled, the user will actually be properly logged-in (because the cookies are enough) and we need an authentication component on Cristal to show that and possibly handle loging out.
- 2: when Cristal is running standalone and accessing an XWiki backend through [cristal-integration-rest-server](https://github.com/xwiki-contrib/cristal-integration/tree/main/cristal-integration-rest-server), the instance does need the OIDC authenticator to be used, and this PR is about solving this specific issue. This is also the reason we can't simply remove the runtime dependency to OIDC, because the first use-case for installing cristal-integration-rest-server is to make the instance accessible to external Cristal instances.

Now, it is true that cristal-integration-rest-server is also a dependency to cristal-integration-resource-handler, which ends up bringing OIDC to instances that might only want to use the internal Cristal. Maybe we can rethink the packages division to avoid that somehow?

manuelleduc commented:
ok, I'd missed the use case where Cristal integration is used headless.
So indeed we need to split in two main modules: one for cristal and it's UI, and on for the server side alone.
OIDC and it's configuration must only be pulled by the "server side alone" module.

pjeanjean committed:
CRISTALINT-13: Automatically set authentication service

* Add headless module

pjeanjean committed:
Merge branch 'main' into CRISTALINT-13

pjeanjean committed:
CRISTALINT-13: Automatically set authentication service

* Bump project to version 1.2.0
* Make REST endpoint to fetch auth service instead

pjeanjean commented:
Instead of the previous fix, the PR now changes the modules architecture so that oidc is no longer a dependency for `cristal-integration-ui`, but only if the new `cristal-integration-headless` is installed. There is also a new REST endpoint so that Cristal can what the check the current authentication service is in https://github.com/xwiki-contrib/cristal/pull/1403

I also bumped the pom versions while I was at it.

pjeanjean committed:
CRISTALINT-13: Automatically set authentication service

* Bump forgotten modules versions

manuelleduc commented:
@pjeanjean imo https://jira.xwiki.org/browse/CRISTALINT-13 title should be updated since we moved from an automatic configuration to a warning to users of a misconfiguration.
Also, I don't see a link to an "upgrade to Cristal 0.24" issue in https://jira.xwiki.org/browse/CRISTALINT-13, which iiuc is going to be required.

pjeanjean commented:
I renamed https://jira.xwiki.org/browse/CRISTALINT-13, and created https://jira.xwiki.org/browse/CRISTALINT-16 because I will need to do it anyway, but the dependencies between issues are not what you think.

This PR adds a REST endpoint completely independent of the Cristal version bundled in the webjar. CRISTALINT-13 is, however, a requirement for https://jira.xwiki.org/browse/CRISTAL-740 so I added a dependency there.
